GDJS/Runtime/scenestack.ts

```diff
@@ -91,13 +100,33 @@ namespace gdjs {
      * Pause the scene currently being played and start the new scene that is specified.
      * If `externalLayoutName` is set, also instantiate the objects from this external layout.
      */
-    push(newSceneName: string, externalLayoutName?: string): gdjs.RuntimeScene {
+    push(
+      newSceneName: string,
+      externalLayoutName?: string
+    ): gdjs.RuntimeScene | null {
       // Tell the scene it's being paused
       const currentScene = this._stack[this._stack.length - 1];
       if (currentScene) {
         currentScene.onPause();
       }
 
+      if (this._runtimeGame.areLayoutAssetsReady(newSceneName)) {
```

Comment:
I guess you can add a comment telling "we could call `loadSceneAssets` and let the loading screen be displayed very briefly even when all assets are ready, but instead we synchronously load the new scene so we don't display an intermediate loading screen during 1 frame."

?